### DISCUSSION TITLE
Dependency versions for git-filter-repo are not correct

### DISCUSSION THREAD

**Discussion by ugultopu** [Homebrew/discussions#776] [Tap maintenance and Homebrew development] [answered]
Although [the project specifies Python 3.5 or greater](https://github.com/newren/git-filter-repo/blob/e9e0308df1e9e792598b8e41d0086d7a785e9cc5/README.md#prerequisites), [the formula specifies a particular version of Python](https://github.com/Homebrew/homebrew-core/blob/6f24624a082d7ce0aba2df6beca47433bf2abeeb/Formula/git-filter-repo.rb).

Likewise, although the project specifies git >= 2.22.0, the formula specifies just Git, without any version restrictions.

Is is not possible to specify a range of versions, just like how the project itself specifies? If no, why not? If yes, then what's the point of forcing a particular dependency version when a range of versions would be sufficient?

Similarly, why doesn't the formula put a restriction on Git version just like the project does?

**Comment by SMillerDev** [comment, marked as answer]
Homebrew will always depend on the latest supported version of a dependency. If no version is specified that means the latest supported version is also the latest available version (except python and openssl where the latest available version has a version number).

Your request for ranges simply doesn't exist in most package managers.

**Reply by ugultopu** [reply]
> If no version is specified

If Homebrew will always depend on the latest version, then what is the point of specifying a version, since it would be ignored I assume?

**Reply by SMillerDev** [reply]
For python we have forced version numbers for maintainer convenience.
